packages/postcss/src/index.ts

```diff
@@ -0,0 +1,306 @@
+              if (i.type === 'ClassSelector' && i.name === '\\-')
+                Object.assign(i, clone(child))
+            })
+            Object.assign(child, parentSelectorAst)
+          })
+          newSelector = generate(prelude)
+        }
+
+        let css = `${newSelector}{${body}}`
+        if (parent)
+          css = `${parent}{${css}}`
+
+        const css_parsed = postcss.parse(css)
+        css_parsed.walkDecls((declaration) => {
+          declaration.source = source
+        })
+        rule.parent.after(css_parsed)
+      }
+      else {
+        const css = postcss.parse(body)
+        css.walkDecls((declaration) => {
+          declaration.source = source
+        })
+        rule.parent.append(css)
+      }
+    }
+
+    rule.remove()
+  })
+}
+function parseTheme(root: Root, uno: UnoGenerator, directiveName: string) {
+  const themeFnRE = new RegExp(`${directiveName}\\((.*?)\\)`, 'g')
+  root.walkDecls((decl) => {
+    const matches = Array.from(decl.value.matchAll(themeFnRE))
+
+    if (!matches.length)
+      return
+
+    for (const match of matches) {
+      const rawArg = match[1].trim()
+      if (!rawArg)
+        throw new Error(`${directiveName}() expect exact one argument, but got 0`)
+
+      let value = uno.config.theme as Record<string, any>
+      const keys = rawArg.slice(1, -1).split('.')
+
+      keys.every((key) => {
+        if (value[key] != null)
+          value = value[key]
+        else if (value[+key] != null)
+          value = value[+key]
+        else
+          return false
+        return true
+      })
+
+      if (typeof value === 'string') {
+        const code = new MagicString(decl.value)
+        code.overwrite(
+          match.index!,
+          match.index! + match[0].length,
+          value,
+        )
+        decl.value = code.toString()
+      }
+      // else if (throwOnMissing) {
+      //   throw new Error(`theme of "${rawArg.slice(1, -1)}" did not found`)
+      // }
+    }
+  })
+}
+
+interface PluginOptions {
+  content?: string[]
+  directiveMap?: {
+    apply: string
+    screen: string
+    theme: string
+  }
+}
+
+function unocss({ content, directiveMap }: PluginOptions = {}) {
+  const fileMap = new Map()
+  const fileClassMap = new Map()
+  const classes = new Set<string>()
+  const config = loadConfig()
+  let uno: UnoGenerator
+  return {
+    postcssPlugin: 'unocss',
+    plugins: [
+      async function (root: Root, result: Result) {
+        const cfg = await config
+
+        if (!cfg.sources.length)
+          throw new Error('UnoCSS config file not found.')
+
+        if (!uno)
+          uno = createGenerator(cfg.config)
+
+        parseApply(root, uno, directiveMap?.apply || 'apply')
+        parseTheme(root, uno, directiveMap?.theme || 'theme')
+        parseScreen(root, uno, directiveMap?.screen || 'screen')
+
+        const entries = await fg(content ?? ['**/*.{html,js,ts,jsx,tsx,vue,svelte,astro}'], {
```

Comment:
May this include/output list be reused?

Reply:
We can't. `fast-glob` expect a `micromatch` pattern, but `include/exclude` options are `picomatch` patterns. The key difference is `picomatch` allow regex, while `micromatch` doesn't.

fast-glob:
- https://www.npmjs.com/package/fast-glob#:~:text=This%20package%20uses%20a%20micromatch%20as%20a%20library%20for%20pattern%20matching.

@rollup/pluginutils: 
- https://www.npmjs.com/package/@rollup/pluginutils#:~:text=A%20valid-,picomatch,-pattern%2C%20or%20array

comparison table: 
- https://github.com/micromatch/picomatch#library-comparisons

Reply:
We can't. fast-glob expect a `micromatch` pattern, but include/exclude options are `picomatch` patterns. The key difference is `micromatch` allow "Brace expansion", while `picomatch` doesn't. And for some reason, fast-glob doesn't allow regexes in the first argument despite `micromatch` supporting it. 

fast-glob:
https://www.npmjs.com/package/fast-glob#:~:text=This%20package%20uses%20a%20micromatch%20as%20a%20library%20for%20pattern%20matching.

@rollup/pluginutils:
https://www.npmjs.com/package/@rollup/pluginutils#:~:text=A%20valid-,picomatch,-pattern%2C%20or%20array

comparison table:
https://github.com/micromatch/picomatch#library-comparisons